src/twisted/mail/test/test_imap.py

```diff
@@ -5348,6 +7242,108 @@ def timedOut(failure):
         return defer.gatherResults([d, self.loopback()])
 
 
+    def test_serverTimesOut(self):
+        """
+        The server times out a connection.
+        """
+        c = Clock()
+        self.server.callLater = c.callLater
+
+        def login():
+            return self.client.login(b'testuser', b'password-test')
+
+        def expireTime():
+            c.advance(self.server.POSTAUTH_TIMEOUT * 2)
+
+        d = self.connected.addCallback(strip(login))
+        d.addCallback(strip(expireTime))
+
+        # The loopback method's Deferred fires the connection is
+        # closed, and the server closes the connection as a result of
+        # expireTime.
+        return defer.gatherResults([d, self.loopback()])
```

Comment:
It is somewhat disheartening to see so many new tests returning `Deferreds` when they really shouldn't need to; however, I can see that it's following the local convention in this test suite, and it's better to have coverage than not.  Can you file a ticket to replace these with `IOPump` and make them not return `Deferred`s?